source/common/listener_manager/listener_manager_impl.cc

```diff
@@ -293,16 +293,22 @@ absl::StatusOr<Network::SocketSharedPtr> ProdListenerComponentFactory::createLis
                                         worker_index);
     };
 
-    auto result = Network::Utility::execInNetworkNamespace(fn, netns.value().c_str());
-
-    // We have a nested absl::StatusOr type, so if there were no issues with changing the namespace,
-    // we want to return the inner absl::StatusOr.
-    if (result->ok()) {
-      return result->value();
+    // Here we're running `fn` in a different network namespace. It will return a `absl::StatusOr`
+    // that wraps the result of the function we pass in, which is another `absl::StatusOr`.
+    auto outer_result = Network::Utility::execInNetworkNamespace(fn, netns.value().c_str());
+
+    // We have a nested absl::StatusOr type. The "outer" result is the result of our attempt to jump
```

Comment:
I guess this could've been:
```
RETURN_IF_NOT_OK_REF(outer_result.status());
return outer_result.value();
```
but I like the additional comments that explain the outer/inner relationship.